### PR TITLE
2.6.0.3

### PR DESCRIPTION
**Pull request by Metalios**
Fixes issues #69 #68 #67 #66 #65 #64 #63 #61 #57 #56

### CODE REVIEW

**Comment by Metalios** [comment]
Oops on the multiple readme updates, hate the github preview

File: Base/Base/Data_Classes/Origin.cs [hunk 1]
```diff
@@ -3,21 +3,21 @@ namespace Base.Data_Classes
...
-            this.Grades = new string[7];
-            this.Grades[0] = "Training";
-            this.Grades[1] = dualO;
-            this.Grades[2] = singleO;
-            this.Grades[3] = "HO";
-            this.Grades[4] = "IO";
-            this.Grades[5] = "IO";
-            this.Grades[6] = "IO";
+            Name = name;
+            Grades = new string[7];
```
Comment (ImaginaryDevelopment) [inline Base/Base/Data_Classes/Origin.cs:13]: this could just be `new [] { "Training", ` ...

File: Base/Base/Data_Classes/Effect.cs [hunk 2]
```diff
@@ -1510,173 +1509,242 @@ public int CompareTo(object obj)
...
+
+                num1 = !(effect.EffectType == Enums.eEffectType.None & EffectType != Enums.eEffectType.None)
```
Comment (ImaginaryDevelopment) [inline Base/Base/Data_Classes/Effect.cs:1527]: holy crap, can this condition not be broken up better into named sub cases for readability?

File: Hero Designer/Data/db/Other/_Mastermind_Pets.Ember_Demon.json [hunk 3]
```diff
@@ -367,177 +367,7 @@
       "Ignore_Buff": [],
       "MutexAuto": true,
       "ForcedClass": "Class_Minion_Henchman",
-      "Effects": [
```
Comment (ImaginaryDevelopment) [inline Hero Designer/Data/db/Other/_Mastermind_Pets.Ember_Demon.json:370]: did all these effects go away on purpose?

File: Hero Designer/Data/db/Other/_Pets.Fallout_Mastermind.json [hunk 4]
```diff
@@ -23,7 +23,7 @@
     {
       "PowerIndex": 6394,
       "PowerSetID": 3088,
-      "StaticIndex": 10762,
```
Comment (ImaginaryDevelopment) [inline Hero Designer/Data/db/Other/_Pets.Fallout_Mastermind.json:26]: why is a static index being changed?

File: Hero Designer/Data/db/Player/Class_Blaster_Blaster_Support.Tactical_Arrow.json [hunk 5]
```diff
@@ -32,7 +32,7 @@
       "ToggleCost": 5.2,
       "PowerIndex": 248,
       "PowerSetID": 27,
-      "StaticIndex": 2885,
```
Comment (ImaginaryDevelopment) [inline Hero Designer/Data/db/Player/Class_Blaster_Blaster_Support.Tactical_Arrow.json:35]: that's a pretty massive static index change, is this intended?

File: Hero Designer/Forms/ImportExportItems/frmImport_Powerset.cs [hunk 6]
```diff
@@ -201,7 +199,7 @@ bool ParseClasses(string iFileName)
                 Exception exception = ex;
                 iStream.Close();
                 int num2 = (int)Interaction.MsgBox(exception.Message, MsgBoxStyle.Critical, "Powerset Class CSV Parse Error");
-                bool flag = false;
+                const bool flag = false;
```
Comment (ImaginaryDevelopment) [inline Hero Designer/Forms/ImportExportItems/frmImport_Powerset.cs:202]: why in the world do we need a `const` false?

File: Hero Designer/Forms/ImportExportItems/frmImport_Powerset.cs [hunk 7]
```diff
@@ -162,36 +162,34 @@ bool ParseClasses(string iFileName)
             {
                 ProjectData.SetProjectError(ex);
                 int num2 = (int)Interaction.MsgBox(ex.Message, MsgBoxStyle.Critical, "Powerset CSV Not Opened");
-                bool flag = false;
+                const bool flag = false;
```
Comment (ImaginaryDevelopment) [inline Hero Designer/Forms/ImportExportItems/frmImport_Powerset.cs:165]: `const` false?

File: Hero Designer/Forms/ImportExportItems/frmImport_Powerset.cs [hunk 8]
```diff
@@ -213,25 +211,23 @@ bool ParseClasses(string iFileName)
         bool ProcessImport()
 
         {
-            bool flag = false;
+            const bool flag = false;
```
Comment (ImaginaryDevelopment) [inline Hero Designer/Forms/ImportExportItems/frmImport_Powerset.cs:214]: const false

File: Hero Designer/Forms/WindowMenuItems/frmDPSCalc.cs [hunk 9]
```diff
@@ -92,145 +91,143 @@ void FillAttackChainWindow(frmDPSCalc.PowerList[] AttackChain)
                 animation += AttackChain[index].Animation;
                 endurance += AttackChain[index].Endurance;
             }
-            Label lblDpsNum = this.lblDPSNum;
+            Label lblDpsNum = lblDPSNum;
             float num = damage / animation;
-            string str2 = num.ToString();
+            string str2 = num.ToString(CultureInfo.CurrentCulture);
```
Comment (ImaginaryDevelopment) [inline Hero Designer/Forms/WindowMenuItems/frmDPSCalc.cs:96]: do we have reasons for this particular culture info selection over the others?

File: Hero Designer/Forms/WindowMenuItems/frmDPSCalc.cs [hunk 10]
```diff
@@ -239,172 +236,169 @@ void lvPower_ItemChecked(object sender, ItemCheckedEventArgs e)
...
                     tbDpsOutput.Text = tbDpsOutput.Text + " -->" + text;
                 }
-                this.GlobalPowerList[this.GlobalPowerList.Length - 1].Damage = !(e.Item.SubItems[2].Text != "-") ? 0.0f : float.Parse(e.Item.SubItems[2].Text);
-                this.GlobalPowerList[this.GlobalPowerList.Length - 1].Endurance = float.Parse(e.Item.SubItems[5].Text);
-                this.GlobalPowerList[this.GlobalPowerList.Length - 1].Recharge = float.Parse(e.Item.SubItems[3].Text);
-                this.GlobalPowerList[this.GlobalPowerList.Length - 1].DamageBuff = float.Parse(e.Item.SubItems[6].Text);
-                this.GlobalPowerList[this.GlobalPowerList.Length - 1].ResistanceDeBuff = float.Parse(e.Item.SubItems[7].Text);
-                this.GlobalPowerList[this.GlobalPowerList.Length - 1].Animation = float.Parse(e.Item.SubItems[4].Text);
-                this.GlobalPowerList[this.GlobalPowerList.Length - 1].RechargeTimer = 0.0f;
+                GlobalPowerList[GlobalPowerList.Length - 1].Damage = !(e.Item.SubItems[2].Text != "-") ? 0.0f : float.Parse(e.Item.SubItems[2].Text);
```
Comment (ImaginaryDevelopment) [inline Hero Designer/Forms/WindowMenuItems/frmDPSCalc.cs:269]: could store a reference to `GlobalPowerList[GlobalPowerList.Length - 1]` in a variable and then just set the props on it, yes?

**Review by ImaginaryDevelopment — COMMENTED** [review]
can't really see what the changes of substance are, buried in the refactorings

**Pull request closed without merging**